openedx/features/pakx/lms/overrides/utils.py

```diff
@@ -418,6 +418,20 @@ def get_course_progress_and_unlock_date(user_id, course_key):
         return None, None
 
 
+def get_or_set_unlock_date(user_id, course_key):
+    """Get unlock date for subsection after setting it if not already set."""
+
+    date_to_unlock, course_progress = get_course_progress_and_unlock_date(user_id, course_key)
+    if not date_to_unlock:
+        return 'date not yet finalized'
+
+    if not course_progress.unlock_subsection_on:
```

Comment:
I would suggest moving this code to `get_course_progress_and_unlock_date` and returning the message from there. WDYT?